### PR TITLE
Use rake-compiler-dock v0.6.0

### PR DESCRIPTION
**Pull request by metaskills**

### CODE REVIEW

**Comment by metaskills** [comment]
I think we should bump the gem version to 1.1 now too.

File: tiny_tds.gemspec [hunk 1]
```diff
@@ -22,7 +22,7 @@ Gem::Specification.new do |s|
   s.add_runtime_dependency     'mini_portile2', '~> 2.0' # Keep this version in sync with the one in extconf.rb !
   s.add_development_dependency 'rake', '~> 10.4'
   s.add_development_dependency 'rake-compiler', '0.9.5'
-  s.add_development_dependency 'rake-compiler-dock', '~> 0.5.1'
+  s.add_development_dependency 'rake-compiler-dock', '0.6.0'
```
Comment (aharpervc) [inline tiny_tds.gemspec]: Is it important to lock to this version specifically, instead of continuing to use `~>`?

Comment (metaskills) [inline tiny_tds.gemspec]: No, I'll change that.

**Comment by metaskills** [comment]
@aharpervc Can you help with this PR? I could not build the Windows gem in this branch.

**Comment by coderjoe** [comment]
@metaskills I just pulled and built on this branch and it seemed to work fine for me. Is there anything I can do to help? I did have to make sure to clear my tmp, pkg, and previous lib directories to allow all the pre-requisite libraries to build.

**Comment by metaskills** [comment]
@coderjoe So `rake gem:windows` worked for you?

**Comment by coderjoe** [comment]
Yep!

**Comment by metaskills** [comment]
OK, let me try again. Can you share the gems you built? Also, just pushed v1.1.0 version bump in this PR.

**Comment by coderjoe** [comment]
I already cleaned my working directory. I'll build again. :)

**Comment by coderjoe** [comment]
What do you know, the build failed on second build... looking into it now.

**Comment by metaskills** [comment]
Cool, pushed up a few changes in anticipation. Nothing major. Thinking of changing this too

```diff
--- a/Rakefile
+++ b/Rakefile
@@ -89,7 +89,7 @@ desc 'Build the windows binary gems per rake-compiler-dock'
 task 'gem:windows' do
   require 'rake_compiler_dock'
   RakeCompilerDock.sh <<-EOT
-    bundle && rake cross native gem RUBY_CC_VERSION=2.0.0:2.1.6:2.2.2:2.3.0 CFLAGS="-Wall"
+    bundle && rake cross native gem RUBY_CC_VERSION=2.4.0:2.3.0:2.2.2:2.1.6:2.0.0 CFLAGS="-Wall"
   EOT
```

**Comment by metaskills** [comment]
Checking out for the night. Lemme know what you find. I'm doing a build locally too. Feel free to commit to this branch too. TTYT.

**Comment by coderjoe** [comment]
It looks like the new version of rake-compiler-dock may have upgraded to a newer version of GCC. This is causing the libiconv build to optimize out an inline function that we need. I'm going to experiment with other explicit optimization settings for libiconv's build and see what happens.

The symptom is an error similar to:

```
/bin/bash ../libtool --mode=link i686-w64-mingw32-gcc -Wl,--export-all-symbols -fPIC -fvisibility=hidden -o libiconv
.la -rpath /home/coderjoe/Code/public/tiny_tds/ports/i686-w64-mingw32/lib -version-info 7:1:5 -no-undefined iconv.lo
 localcharset.lo relocatable.lo iconv-exports.lo libiconv.res.lo
libtool: link: i686-w64-mingw32-gcc -shared  .libs/iconv.o .libs/localcharset.o .libs/relocatable.o .libs/iconv-expo
rts.o .libs/libiconv.res.o    -Wl,--export-all-symbols   -o .libs/libiconv-2.dll -Wl,--enable-auto-image-base -Xlink
er --out-implib -Xlinker .libs/libiconv.dll.a
.libs/iconv.o:iconv.c:(.text+0x1929f): undefined reference to `aliases2_lookup'
.libs/iconv.o:iconv.c:(.text+0x19429): undefined reference to `aliases2_lookup'
.libs/iconv.o:iconv.c:(.text+0x1985e): undefined reference to `aliases2_lookup'
.libs/iconv.o:iconv.c:(.text+0x199ec): undefined reference to `aliases2_lookup'
.libs/iconv.o:iconv.c:(.text+0x1a151): undefined reference to `aliases2_lookup'
collect2: error: ld returned 1 exit status
Makefile:71: recipe for target 'libiconv.la' failed
make[1]: *** [libiconv.la] Error 1
make[1]: Leaving directory '/home/coderjoe/Code/public/tiny_tds/tmp/x86-mingw32/tiny_tds/2.0.0/tmp/i686-w64-mingw32/
ports/libiconv/1.14/libiconv-1.14/lib'
Makefile:33: recipe for target 'all' failed
make: *** [all] Error 2
========================================================================
*** ../../../../ext/tiny_tds/extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.
```

**Comment by coderjoe** [comment]
@metaskills I've pushed a change which seems to have fixed my local build. Do me a favor and give it a go on your machine at your leisure so we can verify it's not just "working for me". :)

**Comment by metaskills** [comment]
Doing that now!

**Comment by aharpervc** [comment]
I am having a problem with `rake gem:windows`, but there's a 95% chance it's just my environment having a problem.

```

Downloading freetds-1.00.21.tar.bz2 (100%)
Extracting freetds-1.00.21.tar.bz2 into tmp/i686-w64-mingw32/ports/freetds/1.00.21... OK
Running git apply with /c/users/aharper/workspace/tiny_tds/ports/patches/freetds/1.00.21/0001-mingw_missing_inet_pton.diff... OK
Running 'configure' for freetds 1.00.21... OK
Running 'compile' for freetds 1.00.21... OK
Running 'install' for freetds 1.00.21... OK
Activating freetds 1.00.21 (from /c/users/aharper/workspace/tiny_tds/ports/i686-w64-mingw32)...
checking for sybfront.h... no
-----
freetds is missing.
Do you have FreeTDS 0.95.80 or higher installed?
-----

```
